### PR TITLE
Adjust regression algorithm

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

This change is due to a request to modify the `OrderImmutabilityRegressionAlgorithm`, before the test only attempted to illegitimately change an order and verified that it was unsuccessful, now we also have an example of using the order ticket to submit an `UpdateOrderRequest`, and then verify that the order was changed later.

#### Related Issue

https://github.com/QuantConnect/Lean/commit/f135fb8060a0ae6cac9d2e399deb8c1c0f368210#commitcomment-42173510

#### Motivation and Context

We are now testing that an order can be legitimately changed and verifying that the change has occurred, which is a good addition to our  set of tests.

#### Requires Documentation Change

None

#### How Has This Been Tested?

Regression algorithm has been debugged and verified the behaviour is as expected.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs [hunk 1]
```diff
@@ -72,37 +85,56 @@ public override void OnOrderEvent(OrderEvent orderEvent)
                 throw new Exception("Orders should be clones, hence not equal!");
             }
 
+            // Try and manipulate orderV2 using the only external accessor BrokerID, since we
+            // are changing a clone the BrokerIDs should not be the same
+            orderV2.BrokerId.Add("FAKE BROKER ID");
+            var orderV3 = Transactions.GetOrderById(orderEvent.OrderId);
+
+            if (orderV2.BrokerId == orderV3.BrokerId)
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs:93]: Here it's actually checking for reference equality, we should verify the lists items, linq has a `SequenceEqual` extension

File: Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs [hunk 2]
```diff
@@ -72,37 +85,56 @@ public override void OnOrderEvent(OrderEvent orderEvent)
...
+                throw new Exception("Quantities should have been updated!");
+            }
 
-            if (orderV2.BrokerId == orderV4.BrokerId)
+            if (updatedOrder.Tag == _originalOrder.Tag)
             {
-                Error("BrokerIDs should not be the same!");
-                throw new Exception("BrokerIDs should not be the same!");
+                Error("Tag should have been updated!");
+                throw new Exception("Tag should have been updated!");
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs:137]: Cool! 👍

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Nice 👍, minor comment

File: Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs (hunk 1 above)
Comment (C-SELLERS) [reply Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs]: Got it!

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**